### PR TITLE
Update HDK to support config structure + use latest jit-engine

### PR DESCRIPTION
**Pull request by alexbaden**

### CODE REVIEW

File: python/pyhdk/_common.pxd [hunk 1]
```diff
@@ -87,16 +88,152 @@ cdef extern from "omniscidb/Shared/SystemParameters.h":
...
-  cdef bool g_enable_dynamic_watchdog
   cdef bool g_enable_debug_timer
 
 cdef extern from "omniscidb/Logger/Logger.h" namespace "logger":
   cdef cppclass CLogOptions "logger::LogOptions":
     CLogOptions(const char*)
 
   cdef void CInitLogger "logger::init"(const CLogOptions &)
+
+cdef extern from "omniscidb/Shared/Config.h":
```
Comment (kurapov-peter) [inline python/pyhdk/_common.pxd:99]: Are there any tools to generate this from plain C structures?

**Comment by alexbaden** [comment]
@kurapov-peter @Garra1980 @ienkovich I removed the pytest changes from this PR so we could get our main branches resynced. I put the pytest work in `alex/pyhdk_github_actions` for now.

File: python/tests/test_pyhdk_sql.py [hunk 2]
```diff
@@ -44,6 +46,6 @@ def test_simple_projection(self):
         assert df["b"].tolist() == [10, 20, 30]
 
     def test_explain(self):
-        res = self.execute_sql("SELECT * FROM test;", just_explain = True)
+        res = self.execute_sql("SELECT * FROM test;", just_explain=True)
         explain_str = res.to_explain_str()
-        assert (explain_str[:15] == "IR for the CPU:")
+        assert explain_str[:15] == "IR for the CPU:"
```
Comment (kurapov-peter) [inline python/tests/test_pyhdk_sql.py:51]: Is this python 2?

Comment (kurapov-peter) [reply python/tests/test_pyhdk_sql.py:51]: Oh it works in 3rd, cool

**Review by kurapov-peter — APPROVED** [review]
LGTM

**Pull request merged**